sys/include/net/gcoap.h

```diff
@@ -46,7 +46,8 @@
  * structs) ordered by the resource path, specifically the ASCII encoding of
```

Comment:
I know this did not change in this PR, but we are touching the doc already so: Where it says "*Internally, gcoap depends on the nanocoap package for base level structs and functionality*" a link to nanocoap could be handy, as it is the first reference to it in the document.

Also, maybe change 'package' to module? as package usually refers to external packages in RIOT.